src/lib/bitbucket-server/bitbucket-server-contributors.ts

```diff
@@ -179,3 +201,114 @@ export const fetchBitbucketReposForProjects = async (
+): Promise<boolean> => {
+  let fileWritten = false;
+  let userPromptedForDetails = false;
+
+  console.log(`Found ${unmonitoredRepoList.length} unmonitored repos`);
+
+  if (repoType.toLowerCase() == 'private') {
+    unmonitoredRepoList = unmonitoredRepoList.filter(
+      (repo) => repo.public == false,
+    );
+  } else if (repoType.toLowerCase() == 'public') {
+    unmonitoredRepoList = unmonitoredRepoList.filter(
+      (repo) => repo.public == true,
+    );
+  }
+  console.log(`Applied the ${repoType} filter to the repo list`);
+  console.log(`Filtered list has ${unmonitoredRepoList.length} repos`);
+
+  for (let u = 0; u < unmonitoredRepoList.length; u++) {
+    const integration = await findIntegrationData(
+      unmonitoredRepoList[u].project.name!,
+      integrations,
+    );
+    if (
+      (integration.integrations == {} || integration.org.id == '') &&
+      (orgID != '' || integrationID != '' || !userPromptedForDetails)
+    ) {
+      const provideDetails = await promptUserForDetails(
+        'Missing Org ID and/or Integration ID for some repos, would you like to provide them? [Y/N] ',
+      );
+      if (provideDetails.toString().toLowerCase() == 'y') {
+        orgID = await promptUserForDetails(
+          'Please provide a valid Snyk Org ID : ',
+        );
+        integrationID = await promptUserForDetails(
+          'Please provide the integration ID for the Org you provided earlier : ',
+        );
+        console.log(
+          `Applying OrgID: ${orgID} and IntegrationID: ${integrationID}`,
+        );
+        debug(`Applying OrgID: ${orgID} and IntegrationID: ${integrationID}`);
+      } else {
+        userPromptedForDetails = true;
+      }
+    }
+    bitbucketServerUnmonitoredReposList.push({
+      integrationId:
+        integration.integrations['bitbucket-server'] || integrationID,
+      orgId: integration.org.id || orgID,
+      target: {
+        projectKey: unmonitoredRepoList[u].project.key,
+        repoSlug: unmonitoredRepoList[u].name,
+      },
+    });
+  }
+  const addParentToJson = { targets: bitbucketServerUnmonitoredReposList };
+  if (!path.endsWith('/')) {
+    path += '/';
+  }
+  fs.writeFile(
+    `${path}Bitbucket-Server-Import-File.json`,
+    JSON.stringify(addParentToJson, null, 4),
+    'utf-8',
+    (err) => {
+      if (err) {
+        console.log(err);
+        fileWritten = false;
+      } else {
+        console.log('Import file created');
+        fileWritten = true;
+      }
+    },
+  );
+  return fileWritten;
+};
+
+const findIntegrationData = async (
+  org: string,
+  integrations: Integration[],
+): Promise<Integration> => {
+  let integration: Integration = {
+    org: { name: '', id: '' },
+    integrations: {},
+  };
+  for (let i = 0; i < integrations.length; i++) {
+    if (integrations[i].org.name == org) {
+      integration = integrations[i];
+    }
+  }
+  return integration;
+};
+
+const promptUserForDetails = async (question: string): Promise<string> => {
```

Comment:
same thing here, move this out to a common code folder/file, like common/utils.ts kindda deal